tests/unit/config/test_configfiles.py

```diff
@@ -134,6 +134,144 @@ def test_yaml_oserror(fake_save_manager, config_tmpdir):
     assert error.traceback is None
 
 
+class TestConfigPyModules:
+
+    """Test for ConfigPy Modules."""
+
+    pytestmark = pytest.mark.usefixtures('config_stub', 'key_config_stub')
+    _old_sys_path = sys.path.copy()
+
+    class ConfPy:
+
+        """Helper class to get a confpy fixture."""
+
+        def __init__(self, tmpdir):
+            self._confpy = tmpdir / 'config.py'
+            self.filename = str(self._confpy)
+
+        def write(self, *lines):
+            text = '\n'.join(lines)
+            self._confpy.write_text(text, 'utf-8', ensure=True)
+
+    class QbModulePy:
+
+        """Helper class to get a QbModulePy fixture."""
+
+        def __init__(self, tmpdir):
+            self._qbmodulepy = tmpdir / 'qbmodule.py'
+            self.filename = str(self._qbmodulepy)
+
+        def write(self, *lines):
+            text = '\n'.join(lines)
+            self._qbmodulepy.write_text(text, 'utf-8', ensure=True)
+
+    @pytest.fixture
+    def confpy(self, tmpdir):
+        return self.ConfPy(tmpdir)
+
+    @pytest.fixture
+    def qbmodulepy(self, tmpdir):
+        return self.QbModulePy(tmpdir)
+
+    def setup_method(self, method):
+        # If we plan to add tests that modify modules themselves, that should
+        # be saved as well
+        TestConfigPyModules._old_sys_path = sys.path.copy()
+
+    def teardown_method(self, method):
+        # Restore path to save the rest of the tests
+        sys.path = TestConfigPyModules._old_sys_path
```

Comment:
Use a fixture instead of `setup_method` / `teardown_method`:

```python
@pytest.fixture(autouse=True)
def restore_sys_path(self):
    old_path = sys.path.copy()
    yield
    sys.path = old_path
```

Reply:
yup, this is a lot cleaner. I probably should learn how to actually use pytest properly...